Common/Data/Consolidators/PeriodCountConsolidatorBase.cs

```diff
@@ -115,6 +117,15 @@ protected PeriodCountConsolidatorBase(Func<DateTime, CalendarInfo> func)
         /// <param name="data">The new data for the consolidator</param>
         public override void Update(T data)
         {
+            if (symbol == null)
+            {
+                symbol = data.Symbol;
+            }
+            else if (symbol != data.Symbol)
+            {
+                throw new Exception($"Previous consolidated symbol ({symbol}) is not the same as in the current data ({data.Symbol})");
```

Comment:
Maybe it would be more precise if instead of using `Exception` we used `InvalidOperationException`, what do you think?

Reply:
Good suggestion.  Done.